node-util/sbin/oo-admin-geartop

```diff
@@ -0,0 +1,188 @@
+    cpu_quota_pct = 100 * cpuquota.to_f / cpuperiod.to_f
+    cg_results  =  cgread(g, 'cpu', 'stat')
+    next if cg_results.nil?
+    throttled_time = cg_results.lines.grep(/throttled_time/)[0].split()[1].to_f / 1_000_000_000
+    cpu_pct = 0
+    if previous.has_key?(g)
+      cpu_pct = (cpu - previous[g].cpu).to_f / (interval * 10_000_000)
+    end
+
+    results[g] = GearStruct.new(g,mem/1024/1024.to_f,memsw/1024/1024.to_f,cpu,
+                        cpu_quota_pct.round(2),throttled_time.round(2),cpu_pct.round(2))
+  end
+
+    return results
+end
+
+# obtain a openshift of uids that have processes currently running
+def get_uids()
+    min_uid = CONF.get('GEAR_MIN_UID')
+    max_uid = CONF.get('GEAR_MAX_UID')
+    district_uuid = Facter.[]('district_uuid').value
+    if district_uuid.casecmp("None") != 0
+        min_uid = Facter.[]('district_first_uid').value unless Facter.[]('district_first_uid').nil?
+        max_uid = Facter.[]('district_max_uid').value unless Facter.[]('district_max_uid').nil?
+    end
+    return %x[ps -eo uid --sort uid].split.uniq & (min_uid..max_uid).to_a
+end
+
+def get_usernames(uids)
+    return uids.collect { |uid| Etc.getpwuid(uid.to_i).name }
+end
+
+def print(strings, fmt_str, limit=nil)
+    limit = strings.size if limit.nil?
+    strings[0..limit].each do |gear|
+      values = gear.to_a
+      values[3] = (values[3].to_f/1_000_000_000)
+      puts fmt_str % values
+    end
+end
+
+if __FILE__ == $0
+    options = {
+      :interval => 3,
+      :iterations => Float::INFINITY,
+      :sort_order => 'c',
+    }
+    OptionParser.new do |opts|
+      opts.banner = "Usage: #{$0} [options]"
+      opts.on("-d N",Integer,"Seconds between iterations") do |d|
+        unless d <= 0
+	  options[:interval] = d
+	end
+      end
+      opts.on('-n N',Integer,"Number of iterations to run") do |n|
+        options[:iterations] = n
+      end
+      opts.on('-s ORDER',[:m, :s, :t, :q, :c],"Sort order: m = mem, s = memsw, t = cpu, q = cpu_quota_pct, c = cpu_pct") do |s|
+        options[:sort_order] = s
+      end
+      opts.on_tail("-h", "--help", "Show this message") do
+        puts opts
+        exit
+      end
+    end.parse!
+
+    previous = {}
+    limit = nil
+    sort = { "m" => :mem,
+             "s" => :memsw,
+             "t" => :cpu,
+             "q" => :cpu_quota_pct,
+             "c" => :cpu_pct
+    }
+    sort_value = sort[options[:sort_order].to_s]
+    interval = options[:interval]
+    iterations = options[:iterations]
+
+    begin
+      0.upto(iterations) do |cnt|
+        loop_start = Time.now
+        uids = get_uids()
+        uuids = get_usernames(uids)
+
+        data = refresh_gear_data(uuids, previous, interval)
+        limit = [ uuids.size, %x[ tput lines ].to_i - 4 ].min
+
+        sort_value = sort['c'] if sort_value.nil?
+        strings = []
+        if cnt > 0
+          strings += data.sort_by do |k,v|
+              v[sort_value.to_sym]
```

Comment:
I can't get this to leak, i think given we're operating on a predefined hash of 5 items we should be safe.